src/app/components/challenge/challenge.component.ts

```diff
@@ -84,7 +84,7 @@ export class ChallengeComponent implements OnInit {
    */
   constructor(@Inject(DOCUMENT) document: any, private router: Router, private route: ActivatedRoute,
               private apiService: ApiService, private globalService: GlobalService,
-              private challengeService: ChallengeService, private authService: AuthService,
```

Comment:
Do we need this change?

Reply:
Yes. It must be public because of the HTML template if conditions. @pushkalkatara

Reply:
In the HTML file the footer varies on wether the user is authenticated or not @pushkalkatara